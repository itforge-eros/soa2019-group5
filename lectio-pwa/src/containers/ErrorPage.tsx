import React, {Component, Fragment} from 'react';
import {Grid} from "@material-ui/core";

type ErrorPageProps = {
	identifier: string,
	title: string,
	body: string
}

class ErrorPage extends Component<ErrorPageProps, any> {
	constructor(props: ErrorPageProps) {
		super(props);
	}

	render() {
		return (
			<Fragment>
				<Grid container direction="column" justify="center" alignItems="center">
					<h1>{ this.props.title }</h1>
					<p>{ this.props.body }</p>
				</Grid>
			</Fragment>
		);
	}
}

export default ErrorPage;
