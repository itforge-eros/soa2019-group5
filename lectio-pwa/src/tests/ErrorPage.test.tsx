import React from 'react';
import renderer from "react-test-renderer";
import ErrorPage from "../containers/ErrorPage";

it('renders ErrorPage successfully', () => {
	const tree = renderer
		.create(
			<ErrorPage identifier="test" title="Sample Error" body="This is a sample error"/>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

export default undefined;
