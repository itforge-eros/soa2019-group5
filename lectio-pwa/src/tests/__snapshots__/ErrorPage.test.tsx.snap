// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ErrorPage successfully 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-direction-xs-column-4 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
>
  <h1>
    Sample Error
  </h1>
  <p>
    This is a sample error
  </p>
</div>
`;
